sdk/go/pulumi-language-go/testdata/projects/l2-resource-keyword-overlap/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"example.com/pulumi-simple/sdk/go/v2/simple"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		_, err := simple.NewResource(ctx, "class", &simple.ResourceArgs{
+			Value: pulumi.Bool(true),
+		})
+		if err != nil {
+			return err
+		}
+		_, err = simple.NewResource(ctx, "export", &simple.ResourceArgs{
+			Value: pulumi.Bool(true),
+		})
+		if err != nil {
+			return err
+		}
+		_, err = simple.NewResource(ctx, "mod", &simple.ResourceArgs{
```

Comment:
Unfortunately this doesn't capture the renaming because we'r not using the resource.

Maybe we can add a stack output for each resource.value, to force Go to generate a variable name.